site/app/controllers/student/TeamController.php

```diff
@@ -461,6 +462,66 @@ public function stopSeekTeam($gradeable_id) {
         $this->core->redirect($return_url);
     }
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/team/setname")
+     */
+    public function setTeamName($gradeable_id): RedirectResponse {
+        $user_id = $this->core->getUser()->getId();
+
+        $gradeable = $this->tryGetGradeable($gradeable_id, false);
+        if ($gradeable === false) {
+            $this->core->addErrorMessage("Invalid gradeable");
+            return new RedirectResponse($this->core->buildCourseUrl());
+        }
+
+        if (!$gradeable->isTeamAssignment()) {
+            $this->core->addErrorMessage("{$gradeable->getTitle()} is not a team assignment");
+            $this->core->redirect($this->core->buildCourseUrl());
```

Comment:
Should return after this if block, otherwise the rest of the function will still execute on development